qutebrowser/components/adblock.py

```diff
@@ -272,8 +285,8 @@ def _merge_file(self, byte_io: io.BytesIO) -> None:
 
         for line in f:
             line_count += 1
-            ok = self._parse_line(line)
-            if not ok:
+            parsed_hosts = self._parse_line(line)
+            if 'error' in parsed_hosts:
```

Comment:
I don't think handling errors in this way is a good idea, I think it might be possible for an 'error' to get in the list in the standard case. 

You could probably handle errors the easiest here using the go way (returning a tuple of 'err, answer')

Reply:
Why not use exceptions?

Reply:
we used the 'return error' method before, and there's no need to spend any time walking the entire stack on every error.

Reply:
Because it didn't need to return any data. Now that it does, an exception is the natural thing to use, and is what's used for the same purpose in a lot of places in qutebrowser. Also, it's not like errors happen often - they are pretty much exceptional :wink: